replicator/src/migrations.rs

```diff
@@ -0,0 +1,63 @@
+use config::shared::{SourceConfig, TlsConfig};
+use sqlx::{
+    postgres::{PgConnectOptions, PgPoolOptions, PgSslMode},
+    Executor,
+};
+
+/// This function runs migrations on the source database when the source database acts
+/// as a state store.
+pub async fn migrate_state_store(source_config: &SourceConfig) -> Result<(), sqlx::Error> {
+    let SourceConfig {
+        host,
+        port,
+        name,
+        username,
+        password,
+        tls,
+    } = &source_config;
```

Comment:
nit: it just adds more code, I would suggest to just inline the calls with `source_config.[param]`.

Reply:
This is removed now since the code is moved into the `SourceConfig::into_connection_config` method and there we don't use the let binding.